### ISSUE TITLE
LinkedIn videos not supported? [Site Request]

### ISSUE THREAD

**Issue by ahm750** [yt-dlp/yt-dlp#1206]
## Checklist

- [x] I'm reporting a new site support request
- [x] I've verified that I'm running yt-dlp version **2021.09.25**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that none of provided URLs violate any copyrights
- [x] The provided URLs do not contain any DRM to the best of my knowledge
- [x] I've searched the bugtracker for similar site support requests including closed ones

## Example URLs

- Single video: https://www.linkedin.com/posts/mishalkhawaja_sendinblueviews-toronto-digitalmarketing-ugcPost-6850898786781339649-mM20

## Description

Is LinkedIn not supported? The reason I ask is that I am able to download LinkedIn Learning videos using a cookie file but not LinkedIn feed videos, like the one I have referenced above.

Here's the verbose log:

```
yt-dlp -J https://www.linkedin.com/posts/fordnation_shubnavratri-hindu-navatri-ugcPost-6852030793657794561-4UPb --verbose
[debug] Command-line config: ['-J', 'https://www.linkedin.com/posts/fordnation_shubnavratri-hindu-navatri-ugcPost-6852030793657794561-4UPb', '--verbose']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] yt-dlp version 2021.09.25
[debug] Python version 3.7.5 (CPython 64bit) - Linux-4.15.0-135-generic-x86_64-with-Ubuntu-18.04-bionic
[debug] exe versions: none
[debug] Optional libraries: Crypto, keyring, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [generic] Extracting URL: https://www.linkedin.com/posts/fordnation_shubnavratri-hindu-navatri-ugcPost-6852030793657794561-4UPb
[generic] fordnation_shubnavratri-hindu-navatri-ugcPost-6852030793657794561-4UPb: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] fordnation_shubnavratri-hindu-navatri-ugcPost-6852030793657794561-4UPb: Downloading webpage
[generic] fordnation_shubnavratri-hindu-navatri-ugcPost-6852030793657794561-4UPb: Extracting information
[debug] [generic] Extracting URL: https://www.linkedin.com/embed/feed/update/urn:li:activity:6852030803648622592?compact=true
[generic] urn:li:activity:6852030803648622592?compact=true: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] urn:li:activity:6852030803648622592?compact=true: Downloading webpage
[generic] urn:li:activity:6852030803648622592?compact=true: Extracting information
ERROR: [generic] Unsupported URL: https://www.linkedin.com/embed/feed/update/urn:li:activity:6852030803648622592?compact=true
Traceback (most recent call last):
  File "/home/bd7/.local/lib/python3.7/site-packages/yt_dlp/extractor/common.py", line 585, in extract
    ie_result = self._real_extract(url)
  File "/home/bd7/.local/lib/python3.7/site-packages/yt_dlp/extractor/generic.py", line 3712, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.linkedin.com/embed/feed/update/urn:li:activity:6852030803648622592?compact=true
Traceback (most recent call last):
  File "/home/bd7/.local/lib/python3.7/site-packages/yt_dlp/extractor/common.py", line 585, in extract
    ie_result = self._real_extract(url)
  File "/home/bd7/.local/lib/python3.7/site-packages/yt_dlp/extractor/generic.py", line 3712, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.linkedin.com/embed/feed/update/urn:li:activity:6852030803648622592?compact=true

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/bd7/.local/lib/python3.7/site-packages/yt_dlp/YoutubeDL.py", line 1227, in wrapper
    return func(self, *args, **kwargs)
  File "/home/bd7/.local/lib/python3.7/site-packages/yt_dlp/YoutubeDL.py", line 1252, in __extract_info
    ie_result = ie.extract(url)
  File "/home/bd7/.local/lib/python3.7/site-packages/yt_dlp/extractor/common.py", line 602, in extract
    e.msg, video_id=video_id, ie=self.IE_NAME, tb=e.traceback, expected=e.expected, cause=e.cause)
yt_dlp.utils.ExtractorError: [generic] Unsupported URL: https://www.linkedin.com/embed/feed/update/urn:li:activity:6852030803648622592?compact=true
```

**Comment by ahm750** [comment]
> I got some linkedin videos to download using this command: `yt-dlp --cookies "C:\Users\username\Downloads\linkedin.com_cookies.txt" -o "C:\Temp\Training Videos\LinkedIn Learning\Powershell\%(playlist_title)s\Lesson %(chapter_number)02d - %(chapter)s\%(playlist_index)s - %(title)s.%(ext)s.mp4" https://www.linkedin.com/learning/powershell-7-essential-training. `
> 
> Here is something that was in my notes from a previous script:
> 
> ```
> https://github.com/ytdl-org/youtube-dl/issues/20326
> 
> Use Chrome to login to linkedin:learning
>     Play a learning video e.g. on this page.
>     Use the cookies plugin to export the file cookies.txt
> 
> How to get the "x-li-identity” header value
>     In Chrome press F12 to open the Network Monitor (development console).
>     Reload the linkedin:learning page with the video.
>     In the Network Monitor, search for “/learning-api”, and click on one of the lines with the HTTP requests.
>     On the right side, scroll down to the request header line “x-li-identity: ...”, and copy 
>     the value to your command line option: --add-header "x-li-identity:..."
> 
> :start
> yt-dlp --cookies "linkedin.com_cookies.txt" --add-header "x-li-identity:dXJuOmxpOmVudGVycHJpc2VQcm9maWxlOih1cm46bGk6ZW50ZXJwcmlzZUFjY291bnQ6MjEzMzg0OSwxMzcwNDk0NjAp" -o "C:\Temp\Training Videos\LinkedIn Learning\%(playlist_title)s\Lesson %(chapter_number)02d - %(chapter)s\%(playlist_index)s - %(title)s.%(ext)s.mp4" https://www.linkedin.com/learning/programming-foundations-databases
> ```
> 
> I'm not sure if the "x-li-identity:" is still needed, as I didn't need it this morning. Hope this helps.

That video is from a LinkedIn Learning course. The issue that I've described is for Linkedin videos that appear in the feed. Those videos cannot be downloaded by yt-dlp at the moment, even when using cookies.

**Comment by cristim** [comment]
@hanimourra I figured out how to download LinkedIn Live videos.

I used the [stream-detector](https://github.com/54ac/stream-detector) Chrome plugin while browsing the LinkedIn live page, and it gave me yt-dlp commands.

I then copied and executed the last of the commands (ending in `livemanifest.ism/manifest(format=m3u8-aapl)"`). and that got me the full video file.

For whatever reason for me the video/audio are slightly out of sync so the lip movement doesn't match the audio but I can live with that for now.
